### PR TITLE
Code Cleanup

### PR DESCRIPTION
**Pull request by dns2utf8**
Frühlingsputz :wink:

### CODE REVIEW

File: src/user_input_compiler.rs [hunk 1]
```diff
@@ -298,7 +291,9 @@ fn match_timesuffix(s :&mut Chars) -> Result<i64, Input> {
   Err( InvalidSyntax(format!("Invalid TimeSuffix")) )
 }
 
-fn starts_with(it :&mut Chars, con :&str) -> bool {
+
+/// Search without modifing the Iterator
+fn starts_with(it :&Chars, con :&str) -> bool {
```
Comment (dbrgn) [inline src/user_input_compiler.rs]: What does `con` stand for? Variable name could be made easier to understand. Same for the parameter in `matches_with`.

Comment (dns2utf8) [inline src/user_input_compiler.rs]: It is the search parameter. I should probably call it `needle`

**Comment by dbrgn** [comment]
"macht" in your commit message should probably be "match". And your capitalization seems a bit random :) In English you only capitalize proper nouns.

**Comment by dns2utf8** [comment]
Cleanup done for today.

File: src/user_input_compiler.rs [hunk 2]
```diff
@@ -219,9 +217,7 @@ fn match_duration(s :&mut Chars) -> Result<Duration,Input> {
 fn match_real(s: &mut Chars) -> Result<f64, Input> {
   let st :String = try!(collect_real(s));
   match st.parse::<f64>() {
-    Ok(val) => {
-      Ok(val)
-    },
+    Ok(val)  => Ok(val),
     Err(msg) => Err( InvalidSyntax(format!("Invalid Real: {:?}", msg)) )
   }
```
Comment (dbrgn) [inline src/user_input_compiler.rs]: In cases like this, where you only do processing in one of the two cases, it's almost always possible to use combinator methods:

``` rust
st.parse::<f64>().map_err(|e| InvalidSyntax(format!("Invalid real: {:?}", e)))
```

File: src/user_input_compiler.rs [hunk 3]
```diff
@@ -245,13 +241,27 @@ fn collect_real(s: &mut Chars) -> Result<String, Input> {
...
+  }
+  
+  
+  
+  let st :String = try!(collect_integer(s));
+  match st.parse::<i64>() {
+    Ok(val)  => Ok(sign * val),
+    Err(msg) => Err( InvalidSyntax(format!("Invalid Integer: {:?}", msg)) ),
+  }
+}
```
Comment (dbrgn) [inline src/user_input_compiler.rs:259]: For all this parsing stuff, you might be better off with a parsing library like https://github.com/Geal/nom. I think it already provides combinators for things like "match an integer".

File: src/user_input_compiler.rs [hunk 4]
```diff
@@ -245,13 +241,27 @@ fn collect_real(s: &mut Chars) -> Result<String, Input> {
...
 fn match_integer(s :&mut Chars) -> Result<i64, Input> {
   let st :String = try!(collect_integer(s));
   match st.parse::<i64>() {
-    Ok(val) => {
-      //s.skip(st.len() -1).next();
-      Ok(val)
-    },
+    Ok(val)  => Ok(val),
     Err(msg) => Err( InvalidSyntax(format!("Invalid Integer: {:?}", msg)) ),
   }
```
Comment (dbrgn) [inline src/user_input_compiler.rs]: Again, you could use `map_err`.

File: src/user_input_compiler.rs [hunk 5]
```diff
@@ -245,13 +241,27 @@ fn collect_real(s: &mut Chars) -> Result<String, Input> {
   Ok( i1 )
 }
 
+fn match_full_integer(s :&mut Chars) -> Result<i64, Input> {
+  let _ = consume_whitespaces(s);
+  let sign = if let Some(c) = s.clone().next() { if c == '-' { -1 } else { 1 } } else { 1 };
```
Comment (dbrgn) [inline src/user_input_compiler.rs]: I'd change this to a multi-line expression or move it out into a function or macro. This one-liner is really hard to read :)

**Comment by dbrgn** [comment]
Good to merge otherwise.

File: src/user_input_compiler.rs [hunk 6]
```diff
@@ -298,18 +305,33 @@ fn match_timesuffix(s :&mut Chars) -> Result<i64, Input> {
   Err( InvalidSyntax(format!("Invalid TimeSuffix")) )
 }
 
-fn starts_with(it :&mut Chars, con :&str) -> bool {
+
+/// Search without modifing the Iterator
+fn starts_with(haystack_iter :&Chars, needle :&str) -> bool {
```
Comment (rnestler) [inline src/user_input_compiler.rs:310]: Whats wrong with [string::starts_with](https://doc.rust-lang.org/std/string/struct.String.html#method.starts_with)?

File: src/user_input_compiler.rs [hunk 7]
```diff
@@ -706,13 +801,35 @@ mod test {
...
   fn webcam_42() {
     match Input::from( format!("/webcam 42") ) {
       WebCam{ nth } => {
         if let Some(nth) = nth { 
-          assert!( if nth == 42 { true } else { info!("wrong Value: {}", nth); false } ) 
+          assert!( if nth == 42 { true } else { println!("wrong Value: {}", nth); false } ) 
         } else {
-          info!("expected OptionalInteger");
+          println!("expected OptionalInteger");
           assert!(false)
```
Comment (rnestler) [inline src/user_input_compiler.rs:833]: Wouldn't it be much easier to just use `assert_eq!(Some(42), nth)`?

File: src/user_input_compiler.rs [hunk 8]
```diff
@@ -595,7 +669,7 @@ mod test {
     match match_timesuffix(&mut s) {
       Ok(v) => assert_eq!(60, v),
       Err(e) => {
-        info!("{:?}", e);
+        println!("{:?}", e);
         assert!(false);
       },
     }
```
Comment (rnestler) [inline src/user_input_compiler.rs:675]: Why not just `assert_eq!(Ok(60), match_timesuffix(&mut s))`?

File: src/user_input_compiler.rs (hunk 6 above)
Comment (dns2utf8) [inline src/user_input_compiler.rs:310]: I pass an Iterator, so i am not sure if it works

File: src/user_input_compiler.rs (hunk 8 above)
Comment (dns2utf8) [inline src/user_input_compiler.rs:675]: Because the PartialEq is not implemented for the Input::\* Types

File: src/user_input_compiler.rs (hunk 7 above)
Comment (dns2utf8) [inline src/user_input_compiler.rs:833]: Working on it: https://github.com/coredump-ch/coredumpbot/pull/26

**Pull request merged**